### ISSUE TITLE
[expo-splash-screen] expo install expo-splash-screen will install old version of package and lead to compilation error

### ISSUE THREAD

**Issue by brunoaduarte** [expo/expo#9432]
### Summary of Issue 

Following the tutorial for expo-splash-screen installation on bare react native projects will lead to compilation errors
https://github.com/expo/expo/tree/master/packages/expo-splash-screen#-installation-in-bare-react-native-projects

`expo install expo-splash-screen` will execute `yarn add expo-splash-screen@^0.3.1`

Then when running "yarn android" it will throw

```
MainActivity.java:19: error: incompatible types: boolean cannot be converted to Class<? extends ViewGroup>
    SplashScreen.show(this, SplashScreenImageResizeMode.CONTAIN, false);
```

If i manually update to the latest 0.4.0 `yarn add expo-splash-screen@~0.4.0` the issue is resolved.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

```
  Expo CLI 3.22.3 environment info:
    System:
      OS: Windows 10 10.0.18362
    Binaries:
      Node: 14.4.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.4 - C:\Users\Myself\AppData\Roaming\npm\yarn.CMD
      npm: 6.14.4 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ~38.0.1 => 38.0.8
      react: ~16.11.0 => 16.11.0
      react-dom: ~16.11.0 => 16.11.0
      react-native: ~0.62.2 => 0.62.2
      react-native-web: ~0.11.7 => 0.11.7
```

Android

### Steps to Reproduce

expo init test -t bare-minimum
cd test
expo install expo-splash-screen
react-native run-android

### Expected Behavior vs Actual Behavior

The expected behavior is for compilation to succeed.

**Comment by OscarYuen** [comment]
For splash screen version lower than 0.4.0, the 3rd argument should be ReactRootView.class
For splash screen version equal to 0.4.0, the 3rd argument should be Boolean.

Some manual steps of the document in master branch has not yet updated.
I updated the step in this PR: https://github.com/expo/expo/pull/9266

**Comment by SimenB** [comment]
Not sure if related, but what version are we supposed to install for managed projects?

This repo says 0.5.0:
- https://github.com/expo/expo/blob/cae1450b51890113d36a3fce3d8b9081f672c200/packages/expo/bundledNativeModules.json#L101
- https://github.com/expo/expo/blob/cae1450b51890113d36a3fce3d8b9081f672c200/packages/expo/package.json#L69

(the latter meaning `yarn add expo` installs `^0.5.0`)

But `expo-cli` says 0.3.1:

![image](https://user-images.githubusercontent.com/1404810/89783045-94c62b00-db16-11ea-971b-33b6fd0703b6.png)


---

```
  Expo CLI 3.23.3 environment info:
    System:
      OS: macOS Mojave 10.14.6
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 12.18.1 - ~/.nvm/versions/node/v12.18.1/bin/node
      Yarn: 1.22.4 - /usr/local/bin/yarn
      npm: 6.14.5 - ~/.nvm/versions/node/v12.18.1/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^38.0.0 => 38.0.9
      react-native: https://github.com/expo/react-native/archive/sdk-38.0.2.tar.gz => 0.62.2
    npmGlobalPackages:
      expo-cli: 3.23.3
```

**Comment by gabrielbezerra81** [comment]
Is there any way to use old version 0.3.1? Expo has it own bundled version 0.5 and running pod-install installs the higher version. Splash screen methods like preventAutoHideAsync and hideAsync only work in 0.3.1.

**Comment by Commoddity** [comment]
> Is there any way to use old version 0.3.1? Expo has it own bundled version 0.5 and running pod-install installs the higher version. Splash screen methods like preventAutoHideAsync and hideAsync only work in 0.3.1.

@gabrielbezerra81 I was having the same issue and I was able to solve it via the following steps:

**Comment by brentvatne** [comment]
> This repo says 0.5.0:

you're looking at the master branch, you should check the sdk-* branch bundledNativeModules.json when looking for the required dependency.

<hr />

that said, i just published a release of the `expo` package to point to `expo-splash-screen@^0.5.0` in `bundledNativeModules` for sdk 38. this will fix the warning that users see when installing `expo-splash-screen` alongside the `expo` package. the original topic of this issue was resolved previously but the fix resulted in this aforementioned warning, and that is now resolved in `expo@38.0.10`.

@gabrielbezerra81 @Commoddity - let's track the issue you're encountering in https://github.com/expo/expo/issues/9286. i pinged @bbarthec about it

**Comment by gabrielbezerra81** [comment]
> > Is there any way to use old version 0.3.1? Expo has it own bundled version 0.5 and running pod-install installs the higher version. Splash screen methods like preventAutoHideAsync and hideAsync only work in 0.3.1.
> 
> @gabrielbezerra81 I was having the same issue and I was able to solve it via the following steps:
> 
> * Install 'expo-splash-screen" at 0.3.1 in package.json
> * Update Podfile as per unimodules instructions to exclude the included version of 'expo-splash-screen' by updating the line `use_unimodules!` to `use_unimodules!(exclude: ['expo-splash-screen'])`
> * Manually install the expo-splash-screen pods from Podfile with the line
> 
> ```
> pod 'EXSplashScreen', :path => '../node_modules/expo-splash-screen/ios'
> ```
> 
> * Keep the AppDelegate.m file as before
> 
> After doing this and running a fresh `pod install` I was able to use the `SplashScreen.preventAutoHideAsync` and `SplashScreen.hideAsync` methods with their desired functionality.

It worked!! Thank you very much, I searched everywhere and found nothing!
